web_server.js

```diff
@@ -48,14 +48,15 @@ io.on('connection', socket => {
 
     // Handle disconnect
     socket.on('disconnect', reason => {
-      log.debug(`${name} (${socket.request.connection.remoteAddress}) has been disconnected. Reason: ${reason}`);
-
       const timestamp = Date.now();
+      const userConnectionDuration = timestamp - socket.meta.joinedAt;
+      log.debug(`${name} (${socket.request.connection.remoteAddress}) has been disconnected after ${userConnectionDuration}. Reason: ${reason}`);
```

Comment:
Please turn the duration into a human readable string. You can install [ms](https://www.npmjs.com/package/ms) for that (make sure to save it to package.json).

Reply:
Nice idea, I have't thought about it. 
Should duration in io.emit('leave', ...) stay in miliseconds?

Reply:
Yes, it should.

Reply:
I was considering using ms, but I'm not sure that will satisfy you. Just look at this simple test:
![mslittletest](https://user-images.githubusercontent.com/14974507/31550063-5dfa7144-b030-11e7-8613-5e2823a9e667.jpg)
Notice that the information about seconds is lost. Similar thing happens, when duration is higher than one hour (e.g duration = 60000*61) then only hours are counted. 
This is not a problem if logs don't have to be so accurate.
I can write functions which will display it with proper format and units, but I'm not sure if you need it and in which file such a funcions such appear.

Reply:
Ok, thank you for telling me @pszczolas :smile: I didn't know that. I found a pretty similar package to `ms`: https://www.npmjs.com/package/pretty-ms
This will do the job :+1: